src/router/views/sta-request.js

```diff
@@ -0,0 +1,280 @@
+import axios from 'axios'
+import moment from 'moment';
+
+const staURL =  process.env.VUE_APP_STA_URL
+const badRequest = 'Something in your request is not correct. It could be the content of the JSON or it could be the server endpoint is not expecting this content'
+const notFound = 'The resource you are looking for does not exist on the system'
+const internalServerError = 'We had a problem with our server. Try again later'
+const serviceUnavailable = 'We’re temporarially offline for maintanance. Please try again later'
+
+const _ = require('lodash');
+
+export default {
+
+    statusCode : 0,
+    requiredAction : 'download',
+    requiredFOI : false,
+    requiredDataStreams: false,
+    requiredObservations: false,
+    requiredSensors: false,
+    requiredObservedProperties: false,
+
+    async getSpecificThing(thingsId, obsrvedProperttyArray, timeRange, selectedFilters,option) {

```

Comment:
This function is too long.  Might be split into several sub-functions

Reply:
Thanks for your feedback, it is done in the new version @kanluo